### PR TITLE
Use new methods for parsing with uninitialized headers from httparse

### LINKED ISSUE #2532: Replace usage of `mem::uninitialized` with `MaybeUninit`
When the following comment was added in #1845, hyper's minimum supported rust version was `1.27.0`

https://github.com/hyperium/hyper/blob/4e9a006498c7bdb5bb2ccb76a4c877f6da7e23b2/src/proto/h1/role.rs#L1-L3

which I assume is why [`std::mem::MaybeUninit`](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html), stabilized in `1.36.0`, could not be used yet. Today, hyper's MSRV is `1.46.0`, so it should now be possible to investigate removing the `#![allow(deprecated)]`.

### PR DESCRIPTION
**Pull request by Soveu**
This patch fixes #2532
It uses methods implemented ~~[in this PR](https://github.com/seanmonstar/httparse/pull/97)~~ [in this PR](https://github.com/seanmonstar/httparse/pull/98), so CI might fail before it gets merged

### CODE REVIEW

File: src/proto/h1/role.rs [hunk 1]
```diff
@@ -1,9 +1,7 @@
-// `mem::uninitialized` replaced with `mem::MaybeUninit`,
-// can't upgrade yet
 #![allow(deprecated)]
```
Comment (seanmonstar) [inline src/proto/h1/role.rs:1]:
```suggestion
```

**Comment by seanmonstar** [comment]
Alright, I've released `httparse` v1.5.0, so we can use it in this branch, and then merge 'er in!

**Comment by Soveu** [comment]
`Some commit messages are invalid` :pensive:

**Comment by Soveu** [comment]
How am I supposed to change commit messages now?

**Comment by seanmonstar** [comment]
Don't worry about the commit messages, I can adjust it when squashing/merging.

**Review by seanmonstar — APPROVED** [review]

**Pull request merged**